lua/swampshop/vgui/customizer.lua

```diff
@@ -296,523 +263,538 @@ local function Container(parent, label)
+    end)
+    --[[
+            vgui("DLabel", function(p)
+                p:SetFont("SS_DESCINSTFONT")
+                p:SetText("WARNING:\nPornographic images or builds are not allowed!")
 
-                    table.sort(v, function(a, b) return a > b end)
-                    SUSL:SetValue(v[2])
+                p.UpdateColours = function(pnl)
+                    pnl:SetTextColor(MenuTheme_TX)
                 end
-            end
 
-            if scale.x == scale.y and scale.y == scale.z then
-                scalebutton:DoClick()
-            end
-        end
+                p:SetColor(SS_SwitchableColor)
+                p:SetContentAlignment(5)
+                p:SizeToContents()
+                p:DockMargin(8, 8, 8, 8)
+                p:Dock(TOP)
+            end)
+            ]]
+    --bottom panel
 
-        --end bunch of copied shit
-        if self.item.configurable.scale_children then
-            CHILDCHECKBOX = CheckboxMaker(pane, "Scale child bones")
-            CHILDCHECKBOX:SetValue(self.item.cfg["scale_children" .. suffix] and 1 or 0)
 
-            CHILDCHECKBOX.OnChange = function(checkboxself, ch)
-                self.item.cfg["scale_children" .. suffix] = ch
-                self:UpdateCfg()
-            end
-        end
-    elseif (self.item.configurable or {}).submaterial then
-        local pane = Container(wear_container, "Skin ID")
-        ATTACHSELECT = vgui.Create("DComboBox", pane)
-
-        for x = 0, math.min(31, (#(LocalPlayer():GetMaterials()) - 1)) do
-            local matname = LocalPlayer():GetMaterials()[x + 1] or ""
-            local exp = string.Explode("/", matname)
-            local nicename = exp[#exp]
-            local sel = tonumber(self.item.cfg.submaterial or 0) == x
-            ATTACHSELECT:AddChoice(tostring(x) .. " (" .. nicename .. ")", tostring(x), sel)
-        end
 
-        ATTACHSELECT.OnSelect = function(panel, index, word, value)
-            self.item.cfg.submaterial = tonumber(value)
-            self:UpdateCfg()
-        end
+end
 
-        ATTACHSELECT:SetWide(200)
-        ATTACHSELECT:Dock(TOP)
-        ATTACHSELECT.Paint = SS_PaintBG
+--gets a value nested into tables using a table of keys. Should be writable?
+local typedef_val = {
+    [TYPE_BOOL] = false,
+    [TYPE_NUMBER] = 0,
+    [TYPE_STRING] = "",
+    [TYPE_VECTOR] = Vector(),
+    [TYPE_ANGLE] = Angle(),
+}
+
+local function typedef(typeid)
+    return typedef_val[typeid]
+end
 
-        ATTACHSELECT.UpdateColours = function(pnl)
-            pnl:SetTextStyleColor(MenuTheme_TX)
-            pnl:SetTextColor(MenuTheme_TX)
-        end
+--we should try to enforce what kind of data is returned by these even if it's already sanitized
+local function checktype(dat, typeid)
+    if typeid == nil then return true end
+    if (istable(typeid)) then return table.HasValue(typeid, TypeID(dat)) end
+
+    return TypeID(dat) == typeid
```

Comment:
you should really just use `type()`

Reply:
it seems generally more sensible to use an enumeration rather than a string in this situation